### DISCUSSION TITLE
I want to set up the hidden server using Nginx in the shadow environment. How should I do?

### DISCUSSION THREAD

**Discussion by beenhuang** [shadow/shadow#1765] [General]
I want to set up the hidden server using Nginx in the shadow environment and implement the following functions:
1. The HS establishes the HS-IP(Introduction Point) circuit and obtains the connection request from the users.
2. After receiving the connection request, the HS establish a circuit with the RP（Rendezvous Point） node to send web page data.
3. The user establishes an client-IP circuit and sends connection requests to the IP node.
4. The user receives a response from the hidden server on the RP Circuit, sends an HTTP request, and receives the web page data from server.

How should I do?

**Comment by robgjansen** [comment]
This discussion has moved to #1767 (please don't open duplicate discussions).
